subprojects/core/src/integTest/groovy/org/gradle/api/internal/tasks/userinput/DefaultBuildScanUserInputHandlerIntegrationTest.groovy

```diff
@@ -16,113 +16,57 @@
 
 package org.gradle.api.internal.tasks.userinput
 
-import org.gradle.api.logging.configuration.ConsoleOutput
-import org.gradle.integtests.fixtures.daemon.DaemonLogsAnalyzer
-import org.gradle.integtests.fixtures.daemon.DaemonsFixture
-import spock.lang.Ignore
 import spock.lang.Unroll
 
 import static org.gradle.integtests.fixtures.BuildScanUserInputFixture.*
 
 class DefaultBuildScanUserInputHandlerIntegrationTest extends AbstractUserInputHandlerIntegrationTest {
 
-    private static final String PLAIN_CONSOLE = 'plain'
-    private static final String RICH_CONSOLE = 'rich'
+    private static final List<Boolean> VALID_BOOLEAN_CHOICES = [false, true]
 
     def setup() {
         file('buildSrc/src/main/java/BuildScanPlugin.java') << buildScanPlugin()
         buildFile << buildScanPluginApplication()
     }
 
     @Unroll
-    def "can ask for license acceptance in interactive build without daemon and #description console"() {
+    def "can ask for license acceptance in interactive build [daemon enabled: #daemon, rich console: #richConsole]"() {
         given:
         interactiveExecution()
-        withConsoleOutput(consoleOutput)
+        withDaemon(daemon)
+        withRichConsole(richConsole)
 
         when:
         def gradleHandle = executer.withTasks(DUMMY_TASK_NAME).start()
 
         then:
-        writeToStdIn(gradleHandle, YES.bytes)
+        writeToStdInAndClose(gradleHandle, YES.bytes)
```

Comment:
❌ Why are we closing stdIn? Maybe that's the whole problem? Shouldn't this be done as part of the teardown process automatically?

Reply:
That's exactly my intuition at the beginning: https://github.com/gradle/gradle-private/issues/975#issuecomment-336710468 . This was proven to be wrong after [Ben's change](https://github.com/gradle/gradle/pull/3222). After this change, we met another `IOException: Write end dead` in addition to the original `Pipe closed`. According to [this post](https://techtavern.wordpress.com/2008/07/16/whats-this-ioexception-write-end-dead/), we should close the write end in time.